components/FilterForm/index.js

```diff
@@ -1,12 +1,31 @@
 import styled from "styled-components";
 import React, { useState } from "react";
-
-export default function FilterForm({ plants, onAddPreference }) {
-  const [plantSize, setPlantSize] = useState("");
-  const [sunlightRequirement, setSunlightRequirement] = useState("");
-  const [waterNeeds, setWaterNeeds] = useState("");
-  const [optimalTemperature, setOptimalTemperature] = useState("");
-  const [petFriendly, setPetFriendly] = useState("");
+import { useRouter } from "next/router";
+
+export default function FilterForm({
+  plants,
+  onAddPreference,
+  initialFilterSettings = {},
+  initialFilterTitle = "",
+  onEditPreference,
+  preferenceId,
+}) {
+  const router = useRouter();
+
+  const initialState = {
+    preferenceTitle: initialFilterTitle,
+    plantSize: initialFilterSettings.plantSize || "",
+    sunlightRequirement: initialFilterSettings.sunlightRequirement || "",
+    waterNeeds: initialFilterSettings.waterNeeds || "",
+    optimalTemperature: initialFilterSettings.optimalTemperature || "",
+    petFriendly: initialFilterSettings.petFriendly || "",
+  };
+
+  const [state, setState] = useState(initialState);
```

Comment:
I think we could work on the naming for the state for a better understanding. 
My suggestions:
- Rename the props `initialFilterSettings` and `initialFilterTitle` to `preferenceFilterSettings` and `preferenceFilterTitle`. These names connect the filter settings with the preference and works good with the other prop names. :)
- Then you can rename `initialState` to `initialPreferenceFilterSettings` for a more descriptive name.
- and the state can be renamed to `settings` as this is the state used inside of your `FilterForm`I would say that is enough for a good understanding (and you are using filterSettings in a good way below already)